src/main/fc/rc.c

```diff
@@ -496,134 +496,182 @@ NOINLINE void initAveraging(uint16_t feedforwardAveraging)
-    float jitterAttenuator = 1.0f;
-    if (pid->feedforwardJitterFactor) {
-        if (rcCommandDeltaAbs < pid->feedforwardJitterFactor) {
-            jitterAttenuator = MAX(1.0f - (rcCommandDeltaAbs + prevRcCommandDeltaAbs[axis]) * pid->feedforwardJitterFactorInv, 0.0f);
-            // note that feedforwardJitterFactorInv includes a divide by 2 to average the two previous rcCommandDeltaAbs values
-            jitterAttenuator = 1.0f - jitterAttenuator * jitterAttenuator;
-        }
-    }
-    prevRcCommandDeltaAbs[axis] = rcCommandDeltaAbs;
 
-    // interpolate setpoint if necessary
-    if (rcCommandDeltaAbs) {
-        // movement!
-        if (prevDuplicatePacket[axis] == true) {
-            rxRate = 1.0f / (rxInterval + prevRxInterval);
-            zeroTheAcceleration = 0.0f;
-            // don't add acceleration, empirically seems better on FrSky
-        }
-        setpointSpeed *= rxRate;
-        prevDuplicatePacket[axis] = false;
-    } else {
-        // no movement!
-        if (prevDuplicatePacket[axis] == false) {
-            // first duplicate after movement
-            setpointSpeed = prevSetpointSpeed[axis];
-            if (fabsf(setpoint) < 0.95f * maxRcRate[axis]) {
-                setpointSpeed += prevAcceleration[axis];
+    float setpointSpeed = 0.0f;
+    float setpointAcceleration = 0.0f;
+    float feedforward = 0.0f;
+
+    if (pid->feedforwardInterpolate) {
+        static float prevRxInterval;
+        // for Rx links which do send duplicate data packets when they send telemetry
+        // interpolate setpointSpeed to minimise steps in feedforward
+        if (rcCommandDeltaAbs) {
+            // movement!
+            // if the previous packet was a duplicate, calculate setpointSpeed over the last two intervals
+            if (prevDuplicatePacket[axis] == true) {
+                rxRate = 1.0f / (rxInterval + prevRxInterval);
             }
-            zeroTheAcceleration = 0.0f; // force acceleration to zero
+            setpointSpeed = setpointDelta * rxRate;
+            prevDuplicatePacket[axis] = false;
         } else {
-            // second and subsequent duplicates after movement should be zeroed
-            setpointSpeed = 0.0f;
-            prevSetpointSpeed[axis] = 0.0f;
-            zeroTheAcceleration = 0.0f; // force acceleration to zero
+            // no movement!
+            if (prevDuplicatePacket[axis] == false) {
+                // interpolate a replacement setpointSpeed value for the first duplicate after normal movement
+                // avoid interpolating when about to hit max deflection
+                if (fabsf(setpoint) < 0.90f * maxRcRate[axis]) {
+                    setpointSpeed = prevSetpointSpeed[axis];
+                    setpointSpeed += prevAcceleration[axis];
+                    // pretend that there was stick movement also, to hold the same jitter value
+                    rcCommandDeltaAbs = prevRcCommandDeltaAbs[axis];
```

Comment:
I think that those lines apply when we had value values for the preceding packet, and then a 'duplicate'.  The 'speed' value to be used when the duplicate arrives doesn't need to be adjusted for the time taken for the duplicate to arrive.  
However, yes, that's a good point, the acceleration element perhaps should be adjusted on the basis of the duplicate interval divided by the preceding interval.  Let me look into this.
From memory, we could have situations where the duplicate interval was long, and if we said, OK now we were accelerating for a long time, so we add a lot of accumulated acceleration to the previous setpoint, and this could be quite wrong.
And since it was uncommon for a duplicate to be meaningfully shorter than a normal packet, I didn't reduce the added acceleration amount when duplicateInterval/previousInterval was < 1.
I'll look into that with a FrSky link at some time in the future.  I do recall that the approach above appeared to work quite well, but I accept it could be improved.